### PR TITLE
# 152807325 Create models and set up associations

### PR DESCRIPTION
**Pull request by kenoseni**
#### What does this PR do ?
Creates models and association from which instances can be created
#### Description of Task to be completed?
Have file migrated to the database with the association defined
#### How should this be manually tested?
Run ./node_modules/.bin sequelize db: migrate from the root folder
#### What are the relevant pivotal tracker stories?
152807325

### CODE REVIEW

File: server/migrations/20171119135234-create-user.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    },
+    createdAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    },
+    updatedAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    },
+  }),
```
Comment (freemile2ru) [inline server/migrations/20171119135234-create-user.js:32]: please do not push migrations files

File: server/migrations/20171119180959-create-center.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    address: {
+      type: Sequelize.STRING
+    },
+    capacity: {
+      type: Sequelize.STRING
+    },
+    createdAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    },
```
Comment (freemile2ru) [inline server/migrations/20171119180959-create-center.js:27]: please do not push migrations files

File: server/migrations/20171119181507-create-event.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    date: {
+      type: Sequelize.DATEONLY
+    },
+    time: {
+      type: Sequelize.STRING
+    },
+    createdAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    },
```
Comment (freemile2ru) [inline server/migrations/20171119181507-create-event.js:24]: please do not push migrations files

File: server/models/center.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        notEmpty: {
+          args: true,
+          msg: 'Name field cannot be empty'
+        },
+        isAlphanumeric: {
+          args: true,
+          msg: 'Allows only alphanumeric characters'
+        }
+      },
+      set(value) {
```
Comment (freemile2ru) [inline server/models/center.js:16]: what is happening here please?

File: server/models/event.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        notEmpty: {
+          args: true,
+          msg: 'Event name cannot be empty'
+        },
+        isAlphanumeric: {
+          args: true,
+          msg: 'Allows only alphanumeric characters'
+        }
+      },
+      set(value) {
```
Comment (freemile2ru) [inline server/models/event.js:16]: what is happening here, please 😳

File: server/models/center.js (hunk 4 above)
Comment (kenoseni) [reply server/models/center.js:16]: Thanks for pointing it out, I will remove this constraint

File: server/models/event.js (hunk 5 above)
Comment (kenoseni) [reply server/models/event.js:16]: Thanks for pointing it out, I will remove this constraint

File: server/migrations/20171119181507-create-event.js (hunk 3 above)
Comment (kenoseni) [reply server/migrations/20171119181507-create-event.js:24]: Thank you for the feedback. I will remove it

File: server/migrations/20171119180959-create-center.js (hunk 2 above)
Comment (kenoseni) [reply server/migrations/20171119180959-create-center.js:27]: Thank you for the feedback. I will remove it

File: server/migrations/20171119135234-create-user.js (hunk 1 above)
Comment (kenoseni) [reply server/migrations/20171119135234-create-user.js:32]: Thank you for the feedback. I will remove it

**Pull request merged**